### DISCUSSION TITLE
Project status and long term outlook

### DISCUSSION THREAD

**Discussion by svenseeberg** [saltstack/salt#67028] [General]
tl;dr 2: https://saltproject.io/blog/2025-01-16-open-hour/

tl;dr 1: If you're interested in supporting a soft fork until Broadcom picks up the work on this repo again, contact me via sven.seeberg@tuerantuer.org.

Hi all,

I was very surprised by the migration from repo.saltproject.io/ to packages.broadcom.com. While the migration itself mostly is not a big issue and can be executed within minutes with the the tools that Salt provides itself, a few things really have me worried:

1. The change of the repository was only announced 1 week in advance. For a project that is supposed to be the foundation for managing infrastructure this is a very short period.

2. packages.broadcom.com does not support IPv6 which repo.saltproject.io did. Its the year 2024 and back end services should be able to live without IPv4. That the project makes a step back here is very confusing. Combined with the short notification period this was a breaking change that creates some challenges for me right now.

3. The project called for community maintainers to become involved: https://saltproject.io/blog/community-maintainers-wanted/. I'm currently pondering the question how I and my organizations can support the project.

4. Most worrying to me is the fact that new commits to the project dropped massively since June: https://github.com/saltstack/salt/graphs/commit-activity

I'm very interested in other/more opinions.

**Comment by OrangeDog** [comment]
I also noticed that these process changes were announced back in January, and have still not been implemented.
https://github.com/saltstack/salt/blob/master/POLICY.rst

**Comment by msvihra** [comment]
I am quite surprised, that even older versions are not available anymore. I get it, versions that are not supported are not safe and should not be used, however, why remove access to older official builds? What is the point? There may be projects, that are in the process of migration to different OS / newer versions of Salt and this just makes everything problematic. I would expect to have old repositories available for some time, to allow us to migrate to supported versions.

**Reply by OrangeDog** [reply]
Not that it's an excuse, but the older versions need different packages for different distros and a bunch of additional dependencies that Salt used to provide.

**Comment by kristoffertt** [comment]
Hey. 
TurtleTraction (www.turtletraction.com) has mirrored saltproject repos. These mirrors are now available to help you smooth the process. 

https://repo.saltproject.io.turtletraction.com/ 
https://archive.repo.saltproject.io.turtletraction.com/ 

Please handle with care and avoid pointing a large number of Salt Minions to it. 
As always, if you have any issues teach out to us on support@turtletraction.com or directly to me here.
Please, share the word in order to avoid interruption of others.

**Reply by msvihra** [reply]
Thanks. I will check and make a backup of the needed archives from there.

**Reply by svenseeberg** [reply]
As [repo.saltproject.io.turtletraction.com/](https://repo.saltproject.io.turtletraction.com/) seems to be no longer available, we set up our own IPv6 mirror of the packages. You can find it on https://salt.verdigado.net.

**Comment by svenseeberg** [comment]
I think it would be nice if Broadcom would transfer the ownership of the brand and this repo to a foundation like the Linux or Apache Foundation. Is anyone in contact with Broadcom? If nothing changes soon, a community fork is needed :-/

**Reply by svenseeberg** [reply]
I'm trying to get into contact with Broadcom. Lets see if they answer.

Beyond that: if anyone is interested in starting/funding a community fork, feel free to reach out to me via sven.seeberg@tuerantuer.org

**Reply by svenseeberg** [reply]
I got some feedback from Broadcom: they did not provide any specifics how soon the project could get back on track. IMHO the situation is a bit problematic as we do not know anymore if security patches are supplied.

**Reply by svenseeberg** [reply]
Who would be interested in a soft fork? IMHO that would encompass merging some open PRs and providing current builds on package servers until this project picks up speed again.

**Comment by jamesharr** [comment]
I wondered my way into here because I noticed the Matrix saltstack bridge has more or less been abandoned and finding the only community link that led somewhere was LIberaChat, which had the following IRC topic set:

> ⓘ Welcome to the remains of #salt. <+> SaltStack has reached End of Life and is no longer supported. <+> So long, and thanks for all the fish.

Guessing that Topic line came from a frustrated employee or community leader. I'd like to continue using SaltStack, but I don't necessarily have the time to make any major contributions.

**Comment by max-arnold** [comment]
https://saltproject.io/blog/2025-01-16-open-hour/

**Reply by OrangeDog** [reply]
> complete confidence

In the UK, that's the phrase used just before a senior minister or government official is fired in disgrace.

**Reply by svenseeberg** [reply]
That is indeed good news. I added the link to the beginning of my initial message.
